libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/factories/NotificationFactory.kt

```diff
@@ -0,0 +1,295 @@
+        // Build the pending intent for when the notification is clicked
+        val openIntent = when {
+            threadId != null -> pendingIntentFactory.createOpenThreadPendingIntent(roomInfo, threadId)
+            else -> pendingIntentFactory.createOpenRoomPendingIntent(roomInfo.sessionId, roomInfo.roomId)
+        }
+
+        val smallIcon = R.drawable.ic_notification
+
+        val channelId = notificationChannels.getChannelIdForMessage(roomInfo.shouldBing)
+        return NotificationCompat.Builder(context, channelId)
+            .setOnlyAlertOnce(roomInfo.isUpdated)
+            .setWhen(lastMessageTimestamp)
+            // MESSAGING_STYLE sets title and content for API 16 and above devices.
+            .setStyle(messageStyle)
+            // A category allows groups of notifications to be ranked and filtered – per user or system settings.
+            // For example, alarm notifications should display before promo notifications, or message from known contact
+            // that can be displayed in not disturb mode if white listed (the later will need compat28.x)
+            .setCategory(NotificationCompat.CATEGORY_MESSAGE)
+            // ID of the corresponding shortcut, for conversation features under API 30+
+            .setShortcutId(roomInfo.roomId.value)
+            // Title for API < 16 devices.
+            .setContentTitle(roomInfo.roomDisplayName)
+            // Content for API < 16 devices.
+            .setContentText(stringProvider.getString(R.string.notification_new_messages))
+            // Number of new notifications for API <24 (M and below) devices.
+            .setSubText(
+                stringProvider.getQuantityString(
+                    R.plurals.notification_new_messages_for_room,
+                    messageStyle.messages.size,
+                    messageStyle.messages.size
+                )
+            )
+            // Auto-bundling is enabled for 4 or more notifications on API 24+ (N+)
+            // devices and all Wear devices. But we want a custom grouping, so we specify the groupID
+            .setGroup(roomInfo.sessionId.value)
+            // In order to avoid notification making sound twice (due to the summary notification)
+            .setGroupAlertBehavior(NotificationCompat.GROUP_ALERT_ALL)
+            .setSmallIcon(smallIcon)
+            // Set primary color (important for Wear 2.0 Notifications).
+            .setColor(accentColor)
+            // Sets priority for 25 and below. For 26 and above, 'priority' is deprecated for
+            // 'importance' which is set in the NotificationChannel. The integers representing
+            // 'priority' are different from 'importance', so make sure you don't mix them.
+            .apply {
+                if (roomInfo.shouldBing) {
+                    // Compat
+                    priority = NotificationCompat.PRIORITY_DEFAULT
+                    /*
```

Comment:
Maybe add a TODO here (and in similar occurrences) explaining why it's commented out? And maybe centralise it/wrap it in a private function.